### PR TITLE
Prevents subjective answer to include tags in maxCharLength

### PR DESCRIPTION
**Pull request by Manyakalra3**
## Summary
- updated the CSS for the `item modal` textarea.
- boxStyling method has been changed so that we may alter attributes straight from props.
- `FormattedAnswer` function has been implemented to  ensure that the answer textarea does not exceed the maximum character limit and to ensure that tags are not counted as part of the maxcharlength.

## TODO

- [x] remove boxStyling prop from itemeditor textarea
- [x] set textarea height to 200 instead of 104 and make it expandable
- [x] change the formattedtext name and make it a method
- [x]  input textbox on player Body.vue in mobile mode should have more height.
- [x] In editor, left side preview pane, subject question selected, increase the default height of the quill editor
- [x] apply green border by default on selected correct option + border doesn't apply on the overall container, make it do so
- [x] indentation in index.css

## Test Plan

- [x] Test Responsiveness
   - [x] Laptop (1200px)
   - [ ] Tablet (760px)
   - [x] Phone (320px)
- [ ] Cross-Browser Testing
   - [x] Chrome
   - [ ] Firefox
   - [ ] Safari (use BrowserStack)
- [ ] ~Local Language Support~
- [ ] Hygiene and Housekeeping
   - [x] Self-review
   - [x] Comments have been added appropriately
   - [ ] Check for bundle size [here](https://bundlephobia.com/) if adding a package
   - [ ] Added relevant details like Labels/Projects/Milestones etc.
   - [ ] ~If adding or removing any environment variable:~
       - [ ] update `docs/ENV.md`
       - [ ] update Github Workflow files
       - [ ] update DockerFile
       - [ ] update the secrets for staging and production
- [ ] Testing
   - [ ] Wrote tests
   - [x] Tested locally
   - [ ] Tested on staging
   - [ ] Tested on an actual physical phone
   - [ ] Tested on production
- [ ] ~Lighthouse Checks~
   - [ ] Images have `alt` attributes
   - [ ] Any `` tags have `width` and `height` specified
   - [ ] Any `target="_blank"` links have `rel="noopener"`
   - [ ] Only SVGs are used as images. If PNGs are used, their size has been optimised.
   - [ ] Any SVGs without text have their `aria-label` attributes set

### CODE REVIEW

File: src/components/Items/Question/Body.vue [hunk 1]
```diff
@@ -343,7 +343,19 @@ export default {
...
+    },
+    formattedAnswer() {
+      // the formatted subjective answer that will be shown on itemModal
+      if (this.subjectiveAnswer == null) {
+        return "";
+      }
+      // added to ensure that answer doesnot exceed maxCharLimit
+      // removes all the HTML tags from the Html String.
+      let regex = /<[^>]*(>|$)|&nbsp;|&zwnj;|&raquo;|&laquo;|&gt;/g;
+      return this.subjectiveAnswer.replace(regex, "");
```
[the remark this answers is not in the document]
Comment (deepansh96) [reply src/components/Items/Question/Body.vue:358]: Ignore above
-- Make a computed variable `isSubjectiveAnswerExceedsMaxLimit` and use that everywhere

File: src/index.css [hunk 2]
```diff
@@ -33,3 +33,6 @@
 .ql-snow.ql-toolbar button.ql-active .ql-fill {
   @apply fill-current text-primary
 }
+.ql-tooltip ,.ql-hidden{
+@apply hidden
```
Comment (deepansh96) [inline src/index.css:37]: indentation

**Review by deepansh96 — CHANGES_REQUESTED** [review]

File: src/components/Items/Question/Body.vue [hunk 3]
```diff
@@ -112,23 +114,50 @@ export default {
...
     return {
       subjectiveAnswer: "", // holds the answer to the subjective question
-      subjectiveBoxHeightLimit: 250, // maximum allowed height of the subjective answer text box in px
+      subjectiveBoxHeightLimit: 200, // maximum allowed height of the subjective answer text box in px
       // set containing the question types in which options are present
       questionTypesSupportingOptions: new Set(["mcq", "checkbox"]),
       isImageLoading: false, // whether the image is loading
       correctOptionClass: "text-white bg-green-500",
       wrongOptionClass: "text-white bg-red-500",
+      // set containing the keys that are enabled when max char limit is reached.
```
Comment (deepansh96) [inline src/components/Items/Question/Body.vue:123]:
```suggestion
      // the keys that need not be filtered when max char limit is reached in a subjective answer
```

File: src/components/Items/Question/Body.vue [hunk 4]
```diff
@@ -112,23 +114,50 @@ export default {
...
+      enabledKeys: [
+        "Meta",
+        "Alt",
+        "Control",
+        "Alt",
```
Comment (deepansh96) [inline src/components/Items/Question/Body.vue:128]: Why is `Alt` two times?

File: src/components/Items/Question/Body.vue [hunk 5]
```diff
@@ -203,6 +232,22 @@ export default {
   },
   components: { Textarea },
   methods: {
+    /** handle any additional text that the user may enter after the character count reaches zero
+     * @param {object} event
+     */
```
Comment (deepansh96) [inline src/components/Items/Question/Body.vue:237]:
```suggestion
    /** 
     * handle any additional text that the user may enter after the character count reaches zero
     * @param {object} event - keydown event containing details of the keys pressed
     */
```

File: src/components/Items/Question/Body.vue [hunk 6]
```diff
@@ -203,6 +232,22 @@ export default {
...
+    handleExtraText(event) {
+      // checks if character limit is reached
```
Comment (deepansh96) [inline src/components/Items/Question/Body.vue:239]: this is not needed
The comment below is enough!

File: src/components/Items/Question/Body.vue [hunk 7]
```diff
@@ -203,6 +232,22 @@ export default {
...
+      // prevents the textarea from accepting keys other than enabled keys when maxcharlimit is reached
+      if (
+        this.charactersLeft == 0 &&
+        // checks whether ctrl key and meta key is pressed when the event occured
```
Comment (deepansh96) [inline src/components/Items/Question/Body.vue:243]:
```suggestion
        // account for shortcuts like ctrl + a / cmd + a
```

**Review by deepansh96 — CHANGES_REQUESTED** [review]

File: src/components/Items/Question/Body.vue (hunk 4 above)
Comment (Manyakalra3) [reply src/components/Items/Question/Body.vue:128]: I will remove this.

File: src/components/Items/Question/Body.vue (hunk 6 above)
Comment (Manyakalra3) [reply src/components/Items/Question/Body.vue]: done

**Pull request merged**